pkg/logs/decoder/matcher.go

```diff
@@ -36,22 +38,27 @@ func (n *NewLineMatcher) SeparatorLen() int {
 
 // BytesSequenceMatcher defines the criterion to whether to end a line based on an arbitrary byte sequence
 type BytesSequenceMatcher struct {
-	sequence []byte
+	sequence  []byte
+	alignment int
 }
 
-// NewBytesSequenceMatcher Returns a new matcher based on custom bytes sequence
-func NewBytesSequenceMatcher(sequence []byte) *BytesSequenceMatcher {
-	return &BytesSequenceMatcher{sequence}
+// NewBytesSequenceMatcher Returns a new matcher based on custom bytes sequence.  Only matches
+// that begin at a multiple of `alignment` are considered.
+func NewBytesSequenceMatcher(sequence []byte, alignment int) *BytesSequenceMatcher {
```

Comment:
Any reason you can't just use `len(sequence)` as the `alignment` vs passing it in? Is there ever a case where they would be different?

Reply:
There are tests that use other lengths for separators.  In general this code is over-generalized and very complex, but I am trying to make minimal changes.

Reply:
Ah I see. Makes sense.